src/common/charts/LegendChart.jsx

```diff
@@ -70,7 +72,7 @@ module.exports = React.createClass({
       <div style={{'width': props.width, 'height': props.height}} >
         {this._renderTitle()}
         {this._renderLegend()}
-        <svg viewBox={props.viewBox} width={props.width - props.sideOffset} height={props.height}>{props.children}</svg>
+        <svg viewBox={this.getViewBox()} width={this.getOuterDimensions().width - props.sideOffset} height={props.height}>{props.children}</svg>
```

Comment:
There is [this approach](http://blog.scottlogic.com/2015/02/02/svg-layout-flexbox.html) using Facebook's css-layout library to implement/emulate a flexbox layout.  However, this comes with a heavy cost when used with React: the background calculations for SVG could only be done once the component has rendered, which not only will drag down performance, but will likely also cause a visible layout change (a la [FOUC](https://en.wikipedia.org/wiki/Flash_of_unstyled_content)) a few hundred microseconds after the initial page render.

I'm not sure how you could do a table or float layout with all the SVG we use for axes.  SVG unfortunately doesn't support any kind of flow layout.  It might be possible to do a table layout using the SVG `foreignObject` but that's not supported by IE.   It might be possible to pull off a table layout if we limit SVG usage to the chart body only, and have the axes labels as non-svg elements.

Ultimately, complex SVG layouts in React is a hard problem, so reducing the use of SVG to the chart body only might be a wise move (indeed, this is why I have the chart title and key in HTML).